File: shop-boot/src/test/java/com/laolang/shop/okhttp/OkHttpBasicTest.java
package com.laolang.shop.okhttp;

import cn.hutool.json.JSONUtil;
import com.laolang.shop.modules.product.req.CategoryAddReq;
import com.laolang.shop.modules.product.req.CategoryEditReq;
import java.io.IOException;
import lombok.extern.slf4j.Slf4j;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import org.testng.Assert;
import org.testng.annotations.Test;

@Slf4j
public class OkHttpBasicTest {

    private static final MediaType JSON = MediaType.get("application/json;charset=utf-8");

    @Test
    public void getTest() throws IOException {
        OkHttpClient client = new OkHttpClient();
        String url = "http://localhost:8090/product/category/list";
        Request request = new Request.Builder().url(url).method("GET", null).build();

        try (Response response = client.newCall(request).execute()) {

            Assert.assertNotNull(response.body());
            String json = response.body().string();
            log.info("request:{}", request);
            log.info("response:{}", response);
            log.info("code:{}", response.code());
            log.info("json:{}", json);
        }
    }

    @Test
    public void putTest() throws IOException {
        OkHttpClient client = new OkHttpClient();
        String url = "http://localhost:8090/product/category/add";
        CategoryAddReq req = new CategoryAddReq();
        req.setParentId(0L);
        req.setName("手机");
        RequestBody body = RequestBody.create(JSON, JSONUtil.toJsonStr(req));
        Request request = new Request.Builder()
                .url(url)
                .put(body)
                .build();

        try (Response response = client.newCall(request).execute()) {

            Assert.assertNotNull(response.body());
            String json = response.body().string();
            log.info("request:{}", request);
            log.info("response:{}", response);
            log.info("code:{}", response.code());
            log.info("json:{}", json);
        }
    }

    @Test
    public void postTest() throws IOException {
        OkHttpClient client = new OkHttpClient();
        String url = "http://localhost:8090/product/category/edit";
        CategoryEditReq req = new CategoryEditReq();
        req.setId(1L);
        req.setParentId(0L);
        req.setName("手机");
        RequestBody body = RequestBody.create(JSON, JSONUtil.toJsonStr(req));
        Request request = new Request.Builder()
                .url(url)
                .post(body)
                .build();

        try (Response response = client.newCall(request).execute()) {

            Assert.assertNotNull(response.body());
            String json = response.body().string();
            log.info("request:{}", request);
            log.info("response:{}", response);
            log.info("code:{}", response.code());
            log.info("json:{}", json);
        }
    }

    @Test
    public void deleteTest() throws IOException {
        OkHttpClient client = new OkHttpClient();
        String url = "http://localhost:8090/product/category/del/1";

        Request request = new Request.Builder().url(url).method("DELETE", null).build();

        try (Response response = client.newCall(request).execute()) {

            Assert.assertNotNull(response.body());
            String json = response.body().string();
            log.info("request:{}", request);
            log.info("response:{}", response);
            log.info("code:{}", response.code());
            log.info("json:{}", json);
        }
    }
}

File: shopmobile/app/src/test/java/com/laolang/shopmobile/OkHttpBasicTest.kt
package com.laolang.shopmobile

import com.google.gson.Gson
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.toRequestBody
import org.junit.Test

class OkHttpBasicTest {

    private val mediaType = "application/json;charset=utf-8".toMediaTypeOrNull()

    @Test
    fun get_test() {
        val client = OkHttpClient.Builder().build()
        val url = "http://localhost:8090/product/category/list"
        val request = Request.Builder().url(url).build()
        val response = client.newCall(request).execute()
        val json = response.body?.string()

        println("request: $request")
        println("response: $response")
        println("code: ${response.code}")
        println("json: $json")
    }

    @Test
    fun put_test() {
        val client = OkHttpClient.Builder().build()
        val url = "http://localhost:8090/product/category/add"
        val req = Gson().toJson(mapOf(Pair("parentId", "0"), Pair("name", "手机")))
        val body: RequestBody = req.toRequestBody(mediaType)
        val request = Request.Builder()
            .url(url)
            .put(body)
            .build()
        val response = client.newCall(request).execute()
        val json = response.body?.string()

        println("request: $request")
        println("response: $response")
        println("code: ${response.code}")
        println("json: $json")
    }

    @Test
    fun post_test() {
        val client = OkHttpClient.Builder().build()
        val url = "http://localhost:8090/product/category/edit"
        val req = Gson().toJson(mapOf(Pair("id", "1"), Pair("parentId", "0"), Pair("name", "手机")))
        val body: RequestBody = req.toRequestBody(mediaType)
        val request = Request.Builder()
            .url(url)
            .post(body)
            .build()
        val response = client.newCall(request).execute()
        val json = response.body?.string()

        println("request: $request")
        println("response: $response")
        println("code: ${response.code}")
        println("json: $json")
    }

    @Test
    fun delete_test() {
        val client = OkHttpClient.Builder().build()
        val url = "http://localhost:8090/product/category/del/1"
        val request = Request.Builder().url(url).delete(null).build()
        val response = client.newCall(request).execute()
        val json = response.body?.string()

        println("request: $request")
        println("response: $response")
        println("code: ${response.code}")
        println("json: $json")
    }

}